### DISCUSSION TITLE
Will this support any openai compliant server?

### DISCUSSION THREAD

**Discussion by delebash** [SemlerPDX/OpenAI-VoiceAttack-Plugin#10] [Ideas]
I am testing running ai on my local machine.  Several llm inference runners such as ollama provide a rest api that emulates openai's.  I haven't loaded this plugin yet but from the docs it looks like you have to supply an openai api key.  As an option we could just point it to an openai compliant server on your local machine.  TY.

**Comment by SemlerPDX** [comment]
This plugin was designed as a means for VoiceAttack users to access the OpenAI API via this plugin for VoiceAttack, and as such, all plugin actions must be accompanied by a valid OpenAI API Key.  Nothing in my plugin attempts to validate that key, that is the job of OpenAI and their API.

You won't be able to target a local machine with my plugin presently, it does one thing and it does it well:  contact the OpenAI API (online) and provide a response to input/requests.  Consideration for local processing will come in time, but not in the current version of my OpenAI API Plugin for VoiceAttack.

Next version is coming in 2025 hopefully first quarter, as many API tools have been deprecated by OpenAI, and they have added new tools (including models) which I'm looking forward to providing for VoiceAttack users.  I am planning on ensuring more flexibility with the next version, including write-in models, and will investigate the possibility of changing the endpoints for API calls to target a local machine, but it likely won't come with the first release of the next version, more realistically further on into 2025 (Spring/Summer if all goes well, god willing and the creeks don't rise).

Thank you very much for your input and for checking out my VoiceAttack plugin!!  Because this is not a proper issue, I am closing this now (and have converted it to a discussion).  Please make use of my Discussion section of this repo for further discourse on this topic.  Cheers!
